app/models/shipit/pull_request.rb

```diff
@@ -163,6 +163,8 @@ def merge!
         commit_message: 'Merged by Shipit',
         merge_method: 'merge',
       )
+      stack.undeployed_commits_count += 1
```

Comment:
You can use before transition blocks: https://github.com/Shopify/shipit-engine/blob/d42d6c5f13bf4102152cbdcc15bf70c7e4a71958/app/models/shipit/pull_request.rb#L87